rust/kernel/error.rs

```diff
@@ -420,6 +420,12 @@ impl From<core::fmt::Error> for Error {
     }
 }
 
+impl From<core::convert::Infallible> for Error {
+    fn from(_: core::convert::Infallible) -> Error {
+        unreachable!()
```

Comment:
You can match on the `Infallible` value. It is an enum with no variants, so you don't need match arms. All (that is all 0) arms return `Error` (or any other type for that matter) so the match itself returns this too. This saves a tiny bit of code to be processed by the compiler.

Reply:
Some new techniques I learned today, thanks @bjorn3 @alex !